data/campaigns/Under_the_Burning_Suns/scenarios/01_The_Morning_After.cfg

```diff
@@ -647,6 +647,7 @@
         [/filter]
 
         # Allow to undo the move before the event is triggered
+        # TODO: remove this its probably unsafe.
```

Comment:
Maybe it could be made safe by unregistering the moveto event handler added on line 668 in `[on_undo]`?

Reply:
Hmm yes we could,  although it would probably be simpler to move the line 668 handler out of there and control it via a variable which is then reset in [on_undo]. Also the outer eents need first_time_only=no for that to work. (so that it can fire egain of redoing)